web/src/katex_server.ts

```diff
@@ -0,0 +1,46 @@
+/* eslint-disable no-console */
+import escape from "escape-html";
+import express from "express";
+import prometheus from "express-prom-bundle";
+import katex from "katex";
+
+const host = "localhost";
+const port = 9700;
+
+const app = express();
+app.use(
+    prometheus({
+        promClient: {collectDefaultMetrics: {}},
+        buckets: [0.0005, 0.001, 0.002, 0.005, 0.01, 0.02, 0.05, 0.1, 0.5],
+        httpDurationMetricName: "katex_http_request_duration_seconds",
+    }),
+);
+
+app.get("/", (req: express.Request, res: express.Response) => {
+    const content = req.query.content ?? "";
+    const is_display = req.query.is_display === "true";
+
+    if (typeof content !== "string") {
+        res.status(400).send("Invalid 'content' argument");
+        return;
+    }
+
+    try {
+        res.send(katex.renderToString(content, {displayMode: is_display}));
+    } catch (error) {
+        if (error instanceof katex.ParseError) {
+            res.status(400).send(escape(error.message));
+        } else {
+            res.status(500).end();
+            if (error instanceof Error) {
+                console.error(error.stack);
+            } else {
+                console.error(error);
+            }
+        }
+    }
+});
+
+app.listen(port, host, () => {
+    console.log(`Server started on http://${host}:${port}`);
+});
```

Comment:
Do we want to add an authentication mechanism to these requests, in order to not have to think about SSRF attacks against the KaTeX rendering system? I guess I don't see any mechanism for that which wouldn't be present already vai the Markdown input feature, so maybe this is best addressed via just having a comment making that argument.

Reply:
I don't think I follow your argument -- can you lay it out?  We don't expose a Markdown-rendering service which isn't ACL'd and rate-limited, so I don't think this is equivalent to that?

Reply:
Yeah I guess it's still a concern; I was mostly thinking about the risk of malicious payloads, and that already exists through putting those payloads directly in the Markdown.

Reply:
Given that, I don't see a reason to not stick a `shared_secret` protection on this, so I intend to do so.